src/utils/k6Client.ts

```diff
@@ -0,0 +1,154 @@
+import { getArch, getPlatform } from '@/utils/electron'
+import { app } from 'electron'
+import { z } from 'zod'
+import path from 'path'
+import { spawn } from 'child_process'
+import readline from 'readline/promises'
+
+const TestOptionsSchema = z.object({
+  cloud: z
+    .object({
+      name: z.string().nullish(),
+      projectID: z.number().nullish(),
+    })
+    .nullish(),
+})
+
+type TestOptions = z.infer<typeof TestOptionsSchema>
+
+const EXECUTABLE_NAME = getPlatform() === 'win' ? 'k6.exe' : 'k6'
+
+const RESOURCES_PATH = MAIN_WINDOW_VITE_DEV_SERVER_URL
+  ? path.join(app.getAppPath(), 'resources', getPlatform())
+  : process.resourcesPath
+
+const EXECUTABLE_PATH = path.join(RESOURCES_PATH, getArch(), EXECUTABLE_NAME)
+
+interface Handlers {
+  onStdOut?: (line: string) => void
+  onStdErr?: (line: string) => void
+  onClose?: (code: number | null) => void
+}
+
+interface SpawnArgs {
+  args: Array<string[] | string | null | undefined | false>
+  env?: NodeJS.ProcessEnv
+}
+
+interface SpawnResult {
+  code: number | null
+  stdout: string[]
+  stderr: string[]
+}
+
+interface ArchiveArgs {
+  scriptPath: string
+  outputPath?: string
+  logFormat?: 'json'
+}
+
+interface InspectArgs {
+  scriptPath: string
+}
+
+export class ArchiveError extends Error {
+  code: number | null
+  stderr: string[]
+
+  constructor(code: number | null, stderr: string[]) {
+    super('Failed to archive script')
+
+    this.code = code
+    this.stderr = stderr
+  }
+}
+
+export class K6Client {
```

Comment:
This is intended to be a typed interface to the k6 CLI. The commands and their options should reflect the options that can be passed to the CLI.